src/main/kotlin/dataProcessor/DataProcessor.kt

```diff
@@ -0,0 +1,7 @@
+package dataProcessor
+
+
+interface DataProcessor<D> {
```

Comment:
This class is very generic, judging by the names. Please consider using more informative names and/or adding an explanation of its purpose.

Reply:
Added comments.